clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -808,7 +808,8 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 }
                 log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
                 this.subscriptions.subscribe(topics, listener);
-                metadata.setTopics(subscriptions.groupSubscription());
+                this.metadata.needMetadataForAllTopics(true);
```

Comment:
@vahidhashemian Feel free to ping me on Google chat if you want to talk about this. It would probably go a little faster that way.